lib/sycamore/sycamore/llms/bedrock.py

```diff
@@ -0,0 +1,70 @@
+import boto3
+import json
+from sycamore.utils.cache import Cache
+from typing import Dict, Optional
+
+
+from sycamore.llms.llms import LLM
+
+DEFAULT_BEDROCK_MODEL = "anthropic.claude-3-5-sonnet-20240620-v1:0"
+DEFAULT_MAX_TOKENS = 1000
+DEFAULT_ANTHROPIC_VERSION = "bedrock-2023-05-31"
+
+
+class Bedrock(LLM):
+    """This is an LLM implementation that uses the AWS Bedrock API to generate text.
+
+    Args:
+        model_name: The name of the Bedrock model to use.
+        cache: A cache object to use for caching results.
+    """
+
+    def __init__(
+        self,
+        model_name: str = DEFAULT_BEDROCK_MODEL,
+        cache: Optional[Cache] = None,
+    ):
+        self._client = boto3.client(service_name="bedrock-runtime")
+        super().__init__(model_name, cache)
+
+    def is_chat_mode(self) -> bool:
+        """Returns True if the LLM is in chat mode, False otherwise."""
+        return True
+
+    def _get_generate_kwargs(self, prompt_kwargs: Dict, llm_kwargs: Optional[Dict] = None) -> Dict:
+        kwargs = {
+            **(llm_kwargs or {}),
+        }
+        if self._model_name.startswith("anthropic."):
```

Comment:
where would one set the temp? We set openai to default 0

Reply:
Updated to be consistent with the OpenAI behavior, though I don't know how I feel about defaulting to 0.